shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -715,6 +715,12 @@ trait LitImpl { self: Lit =>
 }
 
 trait VarImpl { self: Var =>
+  /** Check if the variable name is an integer. */
+  def isIndex: Bool = name.headOption match {
```

Comment:
It seems you could use this in more places where you currently check whether the characters are digit.

Reply:
Absolutely!